### ISSUE TITLE
Create Budget Management API Endpoints

### ISSUE THREAD

**Issue by Limeload** [Limeload/SpendWise#3]
Develop CRUD endpoints for managing Budgets, including creating, updating, viewing, and deleting budgets.

- [ ] Implement CRUD actions in `BudgetsController` (create, read, update, delete budgets).
- [ ] Add strong parameters and validations to Budget model (e.g., name presence).
- [ ] Set up routing for budget endpoints.
- [ ] Add budget serialization for API responses.
- [ ] Write tests for budget management API endpoints.

**Comment by rileychou** [comment]
Hi, can I work on this issue?

**Comment by Limeload** [comment]
@rileychou Yes, you can take some more days. Let me know if you need any help.
